src/OpenDebugAD7/AD7DebugSession.cs

```diff
@@ -2302,34 +2306,88 @@ protected override void HandleSetBreakpointsRequestAsync(IRequestResponder<SetBr
+                            foreach (string condition in conditions)
+                            {
+                                if (LanguageUtilities.IsValidIdentifier(condition))
+                                {
+                                    validConditions.Add(condition);
+                                }
+                                else
+                                {
+                                    m_logger.WriteLine(LoggingCategory.StdErr, string.Format(CultureInfo.CurrentCulture, AD7Resources.Error_Invalid_Exception_Condition, condition));
+                                }
+                            }
+
+                            IEnumerable<string> toAdd = conditions.Except(m_exceptionBreakpoints);
+                            foreach (string condition in toAdd)
+                            {
+                                var exceptionInfo = new EXCEPTION_INFO[1];
+                                exceptionInfo[0].dwState = filter.State;
+                                exceptionInfo[0].guidType = filter.categoryId;
+                                exceptionInfo[0].bstrExceptionName = condition;
+                                m_engine.SetException(exceptionInfo);
+
+                                m_exceptionBreakpoints.Add(condition);
+                            }
+
+                            IEnumerable<string> toRemove = m_exceptionBreakpoints.Except(conditions).ToList();
+                            foreach (string condition in toRemove)
+                            {
+                                var exceptionInfo = new EXCEPTION_INFO[1];
+                                exceptionInfo[0].dwState = filter.State;
+                                exceptionInfo[0].guidType = filter.categoryId;
+                                exceptionInfo[0].bstrExceptionName = condition;
+
+                                m_engine.RemoveSetException(exceptionInfo);
+                                m_exceptionBreakpoints.Remove(condition);
+                            }
+                        }
+                        else
+                        {
+                            SetAllExceptions(filter.State);
+                        }
+                    }
+                    else
                     {
-                        state |= enum_EXCEPTION_STATE.EXCEPTION_STOP_USER_UNCAUGHT;
+                        m_logger.WriteLine(LoggingCategory.StdErr, string.Format(CultureInfo.CurrentCulture, AD7Resources.Error_FilterOption_Not_Supported, filterOption.FilterId));
                     }
-
-                    SetAllExceptions(state);
+                }
+            }
+            else
+            {
+                List<string> filters = responder.Arguments.Filters;
+                if (filters == null || filters.Count == 0)
+                {
+                    SetAllExceptions(enum_EXCEPTION_STATE.EXCEPTION_STOP_SECOND_CHANCE);
```

Comment:
I don't know if you want to fix this now, but we should probably remember which filters in `m_engineConfiguration.ExceptionSettings.ExceptionBreakpointFilters` were NOT referenced, and set those categories to second chance.

Reply:
This path also gets hit when we disable the filter, so wouldn't it also need to include the referenced filter?

Reply:
Right, so what we should do is something create a dictionary from filterName->ExceptionBreakpointFilter. Whenever we find that filter in either `responder.Arguments.FilterOptions` or `responder.Arguments.Filters` we remove it from the dictionary. Any element left in the dictionary is the disabled filters, and they should be set to `enum_EXCEPTION_STATE.EXCEPTION_STOP_SECOND_CHANCE`.

The code works for now, because there is only one filter. But it wouldn't work correctly with multiple.